File: conversions/random-int/elide-random-int.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-random-int

This demonstrates calling the TypeScript random-int implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One random-int implementation shared across TypeScript and Python
- Consistent generate a random integer within a range across services
- No Python custom implementation needed
- Perfect for ID generation, sampling, testing, games
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# random_int_module = require('./elide-random-int.ts')

print("=== Python Calling TypeScript random-int ===\n")

# Example: Basic usage
# result = random_int_module.default()
# print(f"Result: {result}")

print("Real-world use case:")
print("- Python service uses generate a random integer within a range")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent behavior across entire stack")
print("- Perfect for ID generation, sampling, testing, games")
print()

print("Example: Microservices Architecture")
print("┌─────────────────────────────────────┐")
print("│   Elide random-int (TypeScript)         │")
print("│   conversions/random-int/elide-random-int.ts│")
print("└─────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same generate a random integer within a range everywhere!")
print()

print("Problem Solved:")
print("Before: Python custom implementation + JavaScript = inconsistencies")
print("After: One Elide implementation = 100% consistent")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  random_int = require('./elide-random-int.ts')")
print("  # Use it directly!")

File: converted/utilities/random-int/elide-random-int.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-random-int
#
# This demonstrates calling the TypeScript random-int implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One random-int implementation shared across TypeScript and Ruby
# - Consistent generate a random integer within a range across microservices
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript random-int ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# random_int_module = Elide.require('./elide-random-int.ts')

# Example: Basic usage
# result = random_int_module.default()
# puts "Result: #{result}"

puts "Real-world use case:"
puts "- Ruby worker uses generate a random integer within a range"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent behavior across services"
puts "- Perfect for ID generation, sampling, testing, games"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────┐"
puts "│   Elide random-int (TypeScript)         │"
puts "│   conversions/random-int/elide-random-int.ts│"
puts "└─────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same generate a random integer within a range everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby custom + JavaScript = different behaviors"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/random_int.rb"
puts "  RANDOM_INT = Elide.require('./elide-random-int.ts')"
